test/unit/BPool.t.sol

```diff
@@ -954,39 +990,223 @@ contract BPool_Unit_SwapExactAmountIn is BasePoolTest {
+    vm.assume(_tokenIn != VM_ADDRESS);
 
-  function test_Revert_NotBoundTokenOut() private view {}
+    vm.expectRevert('ERR_NOT_BOUND');
+    bPool.swapExactAmountIn(_tokenIn, 0, tokenOut, 0, 0);
+  }
 
-  function test_Revert_NotPublic() private view {}
+  function test_Revert_NotBoundTokenOut(address _tokenOut) public {
+    vm.assume(_tokenOut != VM_ADDRESS);
 
-  function test_Revert_MaxInRatio() private view {}
+    vm.expectRevert('ERR_NOT_BOUND');
+    bPool.swapExactAmountIn(tokenIn, 0, _tokenOut, 0, 0);
+  }
 
-  function test_Revert_BadLimitPrice() private view {}
+  function test_Revert_NotPublic(SwapExactAmountIn_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    _setPublicSwap(false);
 
-  function test_Revert_LimitOut() private view {}
+    vm.expectRevert('ERR_SWAP_NOT_PUBLIC');
+    bPool.swapExactAmountIn(tokenIn, _fuzz.tokenAmountIn, tokenOut, 0, type(uint256).max);
+  }
 
-  function test_Revert_Reentrancy() private view {}
+  function test_Revert_MaxInRatio(SwapExactAmountIn_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    uint256 _tokenAmountIn = bmul(_fuzz.tokenInBalance, MAX_IN_RATIO) + 1;
 
-  function test_Set_InRecord() private view {}
+    vm.expectRevert('ERR_MAX_IN_RATIO');
+    bPool.swapExactAmountIn(tokenIn, _tokenAmountIn, tokenOut, 0, type(uint256).max);
+  }
 
-  function test_Set_OutRecord() private view {}
+  function test_Revert_BadLimitPrice(SwapExactAmountIn_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    uint256 _spotPriceBefore = calcSpotPrice(
+      _fuzz.tokenInBalance, _fuzz.tokenInDenorm, _fuzz.tokenOutBalance, _fuzz.tokenOutDenorm, _fuzz.swapFee
+    );
+    vm.assume(_spotPriceBefore > 0);
 
-  function test_Revert_MathApprox() private view {}
+    vm.expectRevert('ERR_BAD_LIMIT_PRICE');
+    bPool.swapExactAmountIn(tokenIn, _fuzz.tokenAmountIn, tokenOut, 0, _spotPriceBefore - 1);
+  }
 
-  function test_Revert_LimitPrice() private view {}
+  function test_Revert_LimitOut(SwapExactAmountIn_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    uint256 _tokenAmountOut = calcOutGivenIn(
+      _fuzz.tokenInBalance,
+      _fuzz.tokenInDenorm,
+      _fuzz.tokenOutBalance,
+      _fuzz.tokenOutDenorm,
+      _fuzz.tokenAmountIn,
+      _fuzz.swapFee
+    );
 
-  function test_Revert_MathApprox2() private view {}
+    vm.expectRevert('ERR_LIMIT_OUT');
+    bPool.swapExactAmountIn(tokenIn, _fuzz.tokenAmountIn, tokenOut, _tokenAmountOut + 1, type(uint256).max);
+  }
 
-  function test_Emit_LogSwap() private view {}
+  function test_Revert_Reentrancy() public {
+    // Assert that the contract is accesible
+    assertEq(bPool.call__mutex(), false);
 
-  function test_Pull_TokenAmountIn() private view {}
+    // Simulate ongoing call to the contract
+    bPool.set__mutex(true);
 
-  function test_Push_TokenAmountOut() private view {}
+    vm.expectRevert('ERR_REENTRY');
+    bPool.swapExactAmountIn(tokenIn, 0, tokenOut, 0, 0);
```

Comment:
I suggest fuzzing all input arguments in every call, even if those parameters aren't meant to be reached.

When convenient, you could use the `happyPath` regardless of the specific expectation, and then fuzz separately the target variables that are expected to cause a reversion, with the possibility of assuming or bounding them ad hoc; in cases of greater complexity, you could avoid the `happyPath` modifier and rewrite it from within instead (as you did in `test_Revert_MathApprox2`).

```solidity
function test_Revert_SomeError(
    SwapExactAmountIn_FuzzScenario memory _fuzz,
    uint256 _defectiveVariable
) public happyPath(_fuzz) {
    vm.assume(_defectiveVariable > 0);

    bPool.swapExactAmountIn(_fuzz.tokenIn, _fuzz.tokenAmountIn, _fuzz.tokenOut, _defectiveVariable, _fuzz.maxPrice);
}
```